lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -742,7 +742,7 @@ private function getPreAlterTableAlterIndexForeignKeySQL(TableDiff $diff)
         foreach ($diff->changedIndexes as $changedIndex) {
             // Changed primary key
             if ($changedIndex->isPrimary() && $diff->fromTable instanceof Table) {
-                foreach ($diff->fromTable->getPrimaryKeyColumns() as $columnName) {
+                foreach ($diff->fromTable->getPrimaryKeyColumnNames() as $columnName) {
```

Comment:
Why is this not simply `$diff->fromTable->getPrimaryKeyColumns()` and then on line [`746`](https://github.com/doctrine/dbal/pull/2724/files#diff-23f1dda4df961964343780b794ea8095R746) `$columnName->getName()` ?